src/Symfony/Component/Notifier/Bridge/MessageMedia/MessageMediaTransport.php

```diff
@@ -0,0 +1,93 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\MessageMedia;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Adrian Nguyen <vuphuong87@gmail.com>
+ */
+final class MessageMediaTransport extends AbstractTransport
+{
+    protected const HOST = 'api.messagemedia.com';
+
+    private $apiKey;
+    private $apiSecret;
+    private $from;
+
+    public function __construct(string $apiKey, string $apiSecret, ?string $from = null, ?HttpClientInterface $client = null, ?EventDispatcherInterface $dispatcher = null)
+    {
+        $this->apiKey = $apiKey;
+        $this->apiSecret = $apiSecret;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        if (null !== $this->from) {
+            return sprintf('messagemedia://%s?from=%s', $this->getEndpoint(), $this->from);
+        }
+
+        return sprintf('messagemedia://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $endpoint = sprintf('https://%s/v1/messages', $this->getEndpoint());
+        $response = $this->client->request(
+            'POST',
+            $endpoint,
+            [
+                'auth_basic' => $this->apiKey.':'.$this->apiSecret,
+                'json' => [
+                    'messages' => [
+                        [
+                            'destination_number' => $message->getPhone(),
+                            'source_number' => $this->from,
+                            'content' => $message->getSubject(),
+                        ],
+                    ],
+                ],
+            ]
+        );
+
+        if (202 === $response->getStatusCode()) {
+            $result = $response->toArray(false)['messages'][0];
+            $sentMessage = new SentMessage($message, (string) $this);
+            $sentMessage->setMessageId($result['message_id']);
+
+            return $sentMessage;
+        }
+
+        $error = $response->toArray(false);
+
+        throw new TransportException(sprintf('Unable to send the SMS: "%s".', $error['details'][0] ?? $error['message']), $response);
```

Comment:
What if it is a 500 error. There is no json. Then this would lead to type error.

Reply:
You're right @Nyholm . Let me update this one.